core/src/main/java/smashdudes/gameplay/state/playerstate/PlayerState.java

```diff
@@ -31,7 +35,14 @@ public State handleEvent(Event event)
         if (event instanceof StunnedEvent)
         {
             StunnedEvent se = (StunnedEvent)event;
-            return new StunnedState(entity, se.stunTimer);
+            if(event.entity.getComponent(VelocityComponent.class).velocity.y > 0)
+            {
+                return new AirStunnedState(entity, se.stunTimer);
+            }
+            else
+            {
+                return new GroundStunnedState(entity, se.stunTimer);
+            }
```

Comment:
Set the correct stun state depending our current velocity

Reply:
there needs to be a Math.abs here, negative y velocities should not enter the GroundStunnedState